packages/core/src/components/TextField/Mask.jsx

```diff
@@ -266,23 +270,24 @@ Mask.propTypes = {
  * @returns {String}
  */
 export function unmask(value, mask) {
-  if (!value || typeof value !== 'string') return value;
```

Comment:
Previously `unmask` would only return the original value only if `mask` was an invalid option (not currency, phone number, ssn or zip). This rarely happens though, and most of the time the `unmask` function would work as expected, or it would return `''` when a non numeric value was given with a `currency` mask. 

This change makes `maskValue` and `unmask` behave consistently by returning when we cant unmask. This isnt really good either, but it is most similar to the existing behavior while also simplifying the logic.